### PR TITLE
add grain to default pipeline

### PR DESCRIPTION
**Pull request by TrevorDev**
image

- [x] Add ignore to cammelcase issue
- [x] Make dither take amount method, add cos gain part directly in shader
- [x] Use cos instead of softmax
- [x] Change name of controls grainEnabled, grainIntensity, grainAnimated
- [x] move + 1 for animation to outside shader

### CODE REVIEW

File: dist/preview release/typedocValidationBaseline.json [hunk 1]
```diff
@@ -38928,6 +38928,11 @@
           "Comments": {
             "MissingText": true
           }
+        },
+        "GRAIN": {
```
Comment (TrevorDev) [inline dist/preview release/typedocValidationBaseline.json]: Camel case error I cant fix to keep consistency again.

File: src/Shaders/ShadersInclude/helperFunctions.fx [hunk 2]
```diff
@@ -70,9 +70,9 @@ float getRand(vec2 seed) {
 	return fract(sin(dot(seed.xy ,vec2(12.9898,78.233))) * 43758.5453);
 }
 
-vec3 dither(vec2 seed, vec3 color) {
+float dither(vec2 seed, vec3 color, float varianceAmount) {
```
Comment (sebavan) [inline src/Shaders/ShadersInclude/helperFunctions.fx]: seems color is not needed anymore ???

**Comment by sebavan** [comment]
Only one tiny comments and it is in :-)

File: src/Materials/babylon.imageProcessingConfiguration.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ module BABYLON {
     /**
      * Interface to follow in your material defines to integrate easily the
      * Image proccessing functions.
+     * @ignore
```
Comment (TrevorDev) [inline src/Materials/babylon.imageProcessingConfiguration.ts]: @deltakosh do you have any opinion on how to handle these validation errors in the future. Currently we have been ignoring them as they are already all uppercase and modifying them to cammelcase to pass validation might brake back compat,

Comment (deltakosh) [reply src/Materials/babylon.imageProcessingConfiguration.ts]: We will ignore like that:)
All defines do not need doc as they are internal tools mostly

**Pull request merged**